### ISSUE TITLE
Pet Interactions

### ISSUE THREAD

**Issue by WenXin20** [Pokecube-Development/Pokecube-Issues-and-Wiki#586]
#### Issue Description:
- With vanilla_pokemobs enabled, pets have issues with interacting.

#### What happens:
- Pets will not update their sitting state visually, even when pressing the "sit" button.
- Pets cannot be fed/bred.
- Quark allows players to pet dogs by shift-right-clicking, but it opens the gui instead.
- Charm adds random dog textures, but the gui doesn't show the current texture.
- This affects modded pets as well, such as the Hippogryph from Ice and Fire, which I cannot put armor on or feed.

#### What you expected to happen:
- Proper interactions

#### Steps to reproduce:
1. Tame a pet
2. Interact with it.
...

____
#### Affected Versions:

- `Pokecube AIO: v3.5.1`
- `Minecraft: v1.16.4`
- `Forge: v35.1.15`

**Comment by Thutmose** [comment]
hmm, any suggestions for how to specify opening pokemob inventory vs accessing them normally? for trainers I made it via right clicking with a pokedex

**Comment by WenXin20** [comment]
Right-clicking with a Pokedex sounds best. Or maybe a shift-right-click, although that also might conflict with some mods.

**Comment by Thutmose** [comment]
hmm, I can look into adjustments so right click with pokedex (but not watch) will apply the interaction for non-standard pokemobs

**Comment by WenXin20** [comment]
Alright, that sounds good.

**Comment by Thutmose** [comment]
that commit should have dealt with this, I can probably have a build out tomorrow or so, if you want to try it earlier, you can probably build it by downloading the repository and running the build.bat, it should make a compiled jar to test with.

The changes make it so that if you right click with a non-standard pokemob, it will do the following:

check for berry use/etc, if so apply those
check if holding a pokedex, if so, open gui
otherwise do normal interaction

**Comment by Thutmose** [comment]
Can you update the description for this for which are still a problem? I can probably try to look into the remaining next.

**Comment by WenXin20** [comment]
Alright, it's been updated.

**Comment by WenXin20** [comment]
I also realized that if a pet is tamed first, then I can't use a pokecube to store it.

**Comment by Thutmose** [comment]
any links for charm source for how it "adds random dog textures"? the code currently tries to call `writeAdditional` and `readAdditional` to sync things like that over...

**Comment by Thutmose** [comment]
also, when I tamed a cat with fish, it magically found a pokecube to recall into

**Comment by Thutmose** [comment]
oh, I see, WolfEntity is one of the ones that doesn't support `readAdditional` on client side

**Comment by WenXin20** [comment]
I think this is it: https://github.com/svenhjol/CharmForge/blob/master/src/main/java/svenhjol/charm/client/VariantMobTexturesClient.java

That's strange. I pressed the recall keybind and the pets don't return unless they were caught with a pokeball.

**Comment by Thutmose** [comment]
how many are fixed here? https://github.com/Pokecube-Development/Pokecube-Issues-and-Wiki/releases/tag/1.16.4-3.6.1

**Comment by WenXin20** [comment]
It looks like everything here is fixed. Although I just noticed that when I try to jump while on a horse, it'll jump before I let go of the space bar, so a full powered jump doesn't seem to be possible.

**Comment by Thutmose** [comment]
ahh right, I need to make a check somehow for "vanilla" control behaviour, and then if that is found, disable the custom stuff.
